### DISCUSSION TITLE
Unable to add hotspots on point cloud model

### DISCUSSION THREAD

**Discussion by virtualheritage** [google/model-viewer#4217] [Q&A] [answered]
I would like to add some annotations on a point cloud model, but it seems that click event doesn't get triggered on this type of models. I have tried in modelviewer editor, but no success.

Validation report shows that this model has 0 triangles, so I guess there is no surface where you can click thus click event doesn't get triggered.

Any workaround or idea to allow click events being triggered when clicking vertices?

**Comment by elalish** [comment]
I'm not sure. I would recommend falling back to setting `data-position` to world-coordinate values by hand.

**Reply by virtualheritage** [reply]
Thanks for your answer, but sadly our use-case requires non-programmer users being able to add hotspots, so that solution wouldn't be suitable for us.

**Reply by elalish** [reply]
Well, it is possible in three.js: https://threejs.org/examples/#webgl_interactive_raycasting_points and it looks like a bug in our `positionAndNormalFromPoint()` method that's discarding them. Would you mind linking a point-cloud GLB for me to test with? It still won't work in our editor since that uses `surfaceFromPoint()` now, but it should be pretty easy to set up your own hotspot-adding page.

**Reply by virtualheritage** [reply]
Of course: _removed link_

**Reply by elalish** [reply, marked as answer]
Okay, I've got this fixed over in #4228. I tested that it works on your model, though it often picks points that don't seem right. My best guess is that model has points that are colored transparent (alpha < 1). I'd recommend removing those points, or changing their colors.

**Reply by virtualheritage** [reply]
Thanks!
